From 8949aaaeae7d6c958d6e3eecfe7f9519bf1aeb66 Mon Sep 17 00:00:00 2001
From: ruslanbay <67730802+ruslanbay@users.noreply.github.com>
Date: Wed, 10 Dec 2025 15:08:31 +0100
Subject: [PATCH 18/56] switch to multiplanar video capture type

The device is advertised[1] as supporting multiplanar capture, which libcamera will assess[2] and decide that it should look for buffers with the type V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE, but .enum_fmt() returns no formats if the passed in buffer type[3] is not V4L2_BUF_TYPE_VIDEO_CAPTURE (note the lack of _MPLANE).

Change that line to V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE.

- [1] https://github.com/ruslanbay/linux/blob/fcab57e77edf3944f80c5fbe21ec344402a7fd4d/drivers/media/pci/intel/ipu-isys-video.c#L395
- [2] https://github.com/libcamera-org/libcamera/tree/v0.6.0/src/libcamera/v4l2_videodevice.cpp#L611
- [3] https://github.com/ruslanbay/linux/blob/fcab57e77edf3944f80c5fbe21ec344402a7fd4d/drivers/media/pci/intel/ipu-isys-video.c#L427
---
 drivers/media/pci/intel/ipu-isys-video.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/ipu-isys-video.c b/drivers/media/pci/intel/ipu-isys-video.c
index 5a8c8d22f877..c54813e7b8ce 100644
--- a/drivers/media/pci/intel/ipu-isys-video.c
+++ b/drivers/media/pci/intel/ipu-isys-video.c
@@ -439,7 +439,7 @@ int ipu_isys_vidioc_enum_fmt(struct file *file, void *fh,
 		if ((ipu_isys_pfmts[i].is_meta &&
 		     f->type != V4L2_BUF_TYPE_META_CAPTURE) ||
 		    (!ipu_isys_pfmts[i].is_meta &&
-		     f->type != V4L2_BUF_TYPE_VIDEO_CAPTURE))
+		     f->type != V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE))
 			continue;
 
 		if (f->mbus_code && f->mbus_code != ipu_isys_pfmts[i].code)
-- 
2.51.0

